scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+package org.oppia.android.scripts.maven
+
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import org.oppia.android.scripts.maven.maveninstall.MavenListDependency
+import org.oppia.android.scripts.maven.maveninstall.MavenListDependencyTree
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.OriginOfLicenses
+import org.oppia.android.scripts.proto.PrimaryLinkType
+import org.oppia.android.scripts.proto.SecondaryLinkType
+import java.io.File
+import java.io.FileInputStream
+import java.net.URL
+import java.util.concurrent.TimeUnit
+
+private const val WAIT_PROCESS_TIMEOUT_MS = 60_000L
+private const val LICENSES_TAG = "<licenses>"
+private const val LICENSES_CLOSE_TAG = "</licenses>"
+private const val LICENSE_TAG = "<license>"
+private const val NAME_TAG = "<name>"
+private const val URL_TAG = "<url>"
+
+/**
+ * Script to compile the list of the third-party Maven dependencies (direct and indirect both)
+ * on which Oppia-android depends.
+ *
+ * Usage:
+ *   bazel run //scripts:generate_maven_dependencies_list  -- <path_to_directory_root>
+ *   <path_to_maven_install_json> <path_to_maven_dependencies_textproto>
```

Comment:
Why is the path to the textproto necessary? It seems like we always use the same file since we pull it in during build time.

Reply:
I am passing it as a parameter for testing purposes.

Reply:
@prayutsu I think now these paths can be used directly as in scripts instead of passing them.

Reply:
If you update this, make sure to update PR description and below example too.

Reply:
The problem here is that even if we create new pb files during testing when we'll run the script, it will still read the file at 
`scripts/assets/maven_dependencies.pb` as we are going to hard code it in the `getProto()` function. We'll not be able to control it and it will always read the updated list which will not enable us to test different scenarios.

Reply:
@BenHenning @rt4914 
The problem here is that even if we create new pb files during testing when we'll run the script, it will still read the file at 
`scripts/assets/maven_dependencies.pb` as we are going to hard code it in the `getProto()` function. We'll not be able to control it and it will always read the updated list which will not enable us to test different scenarios.

Reply:
I think I clarified this in our chat, but please let me know if additional clarity is needed.

Reply:
@BenHenning I tried mkdiring only directories, but it also gives the same error - 
![Screenshot from 2021-07-15 12-56-46](https://user-images.githubusercontent.com/54636525/125747210-9325fa5f-ce6a-444f-b1bb-7dc80d5a90cc.png)

Also tried this as per https://stackoverflow.com/a/3634906/12215015
![Screenshot from 2021-07-15 13-00-47](https://user-images.githubusercontent.com/54636525/125747577-6bbcef06-4326-4a10-a70e-50ddd33eec11.png)

But didn't work.

Reply:
As discussed, we root caused the issue during the team chat & found a workaround.